de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/joiner/JoinerNodeModel.java

```diff
@@ -418,19 +452,37 @@ protected void loadJsonSetting() throws IOException, CanceledExecutionException
     if (nodeSettings.secondModelParameters != null) {
       representation.setSecondModelParameters(nodeSettings.secondModelParameters);
     }
-
-
-    if (!JoinerNodeUtil.parametersNeedUpdate(newFirstModelParameters, JoinerNodeModel.SUFFIX_FIRST, JoinerNodeModel.SUFFIX_SECOND)) {
-      representation.updateParameters(newFirstModelParameters, newSecondModelParameters);
-      viewValue.updateParameters(newFirstModelParameters, newSecondModelParameters);
-    } else {
-      viewValue.modelMetaData = nodeSettings.modelMetaData;
+    if (firstInputPort != null) {
+      List<Parameter> newFirstModelParameters =
+          SwaggerUtil.getParameter(firstInputPort.modelMetadata);
+      List<Parameter> newSecondModelParameters =
+          SwaggerUtil.getParameter(secondInputPort.modelMetadata);
+      if (!JoinerNodeUtil.parametersNeedUpdate(newFirstModelParameters,
+          JoinerNodeModel.SUFFIX_FIRST, JoinerNodeModel.SUFFIX_SECOND)) {
+        representation.updateParameters(newFirstModelParameters, newSecondModelParameters);
+        viewValue.updateParameters(newFirstModelParameters, newSecondModelParameters);
+      } else {
+        viewValue.modelMetaData = nodeSettings.modelMetaData;
+      }
     }
 
     representation.setSecondModelViz(visualizationScript);
 
   }
 
+  /*
+   * A migration method, takes an old setting file and extract the data inside as list of
+   * de.bund.bfr.metadata.swagger.parameter
+   */
+  private static List<Parameter> getModelParametersfromOldFile(File configFile)
+      throws FileNotFoundException, IOException, ParseException {
+    JSONParser parser = new JSONParser();
+    Object obj = parser.parse(new FileReader(configFile));
+    JSONObject jsonObject = (JSONObject) obj;
+    JSONArray parameterList = (JSONArray) jsonObject.get("parameter");
+    return MAPPER.readValue(parameterList.toJSONString(), new TypeReference<List<Parameter>>() {});
+  }
```

Comment:
The biggest problem in this method is the leak of the FileReader that is not closed (480).

This method is similar to creatRelationList of the 1.7.2 JoinerNodeModel. We are using 2 JSON libraries. First we parsed the passed configuration file with JSON simple, get the parameters, convert them to a string which is finally deserialised to a Parameter objects with Jackson.

1. How does this JSON configuration file look like? Can we deserialise with data binding to a POJO? If this does not exist we can just create it and drop this whole method.
2. In case 1 takes a lot of effort we could do everything in this method with Jackson. The contents of the file can be read as a JsonTree and its parameters can be deserialised manually. [treeToValue](https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectMapper.html#treeToValue(com.fasterxml.jackson.core.TreeNode,%20java.lang.Class))

We should not forget to close the FileReader. With Jackson we would not need it anyway https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectMapper.html#readTree(java.io.File).